improver_tests/psychrometric_calculations/hail_size/test_hail_size.py

```diff
@@ -0,0 +1,314 @@
+        name="relative_humidity_on_pressure_levels",
+        units="kg/kg",
+        attributes=LOCAL_MANDATORY_ATTRIBUTES,
+    )
+
+    return humidity_cube
+
+
+def metadata_check(hail_cube):
+    """Checks the hail cube produced by plugin has the expected metadata."""
+    assert hail_cube.long_name == "size_of_hail_stones"
+    assert hail_cube.units == "mm"
+    assert hail_cube.dtype == np.int8
+
+    coord_names = [coord.name() for coord in hail_cube.coords()]
+    assert coord_names == [
+        "realization",
+        "latitude",
+        "longitude",
+        "forecast_period",
+        "forecast_reference_time",
+        "time",
+    ]
+    assert hail_cube.shape == (2, 3, 2)
+    attributes = [attr for attr in hail_cube.attributes]
+    if "mosg__model_configuration" in attributes:
+
+        assert attributes == [
+            "title",
+            "source",
+            "institution",
+            "mosg__model_configuration",
+        ]
+    else:
+        assert attributes == ["title", "source", "institution"]
+
+
+"""literiture tephigram link
+Fawbush, E.J., and R.C. Miller. 1953.
+“A method for forecasting hailstone size at the earth’s surface.”
+Bulletin of the American Meteorological Society 34: 235-244.
+https://doi.org/10.1175/1520-0477-34.6.235
+"""
+
+
+@pytest.mark.parametrize(
+    "ccl_p,ccl_t,humidity,expected",
+    (
+        (75000, 290, 0.001, 20,),  # values approx from tephigram in literiture
+        (94000, 300, 0.001, 0),  # vertical value negative
+        (1000, 360, 0.001, 0),  # horizontal value negative
+        (95000, 330, 0.001, 80),  # vertical greater than length of table
+        (150000, 350, 0.1, 20),  # horizontal greater than length of table
+        (75000, 265, 0.001, 0),  # ccl temperature below 268.15
+    ),
+)
+def test_basic_hail_size(
+    ccl_pressure,
+    ccl_temperature,
+    temperature_on_pressure_levels,
+    relative_humidity_on_pressure,
+    ccl_p,
+    ccl_t,
+    humidity,
+    expected,
+):
+    """Tests the hail_size plugin with values for ccl temperature, ccl pressure
+    and relative humidity to check for expected result. Also checks the metadata
+    of the produced hail_size cube"""
+    ccl_pressure.data = np.full_like(ccl_pressure.data, ccl_p)
+    ccl_temperature.data = np.full_like(ccl_temperature.data, ccl_t)
+    relative_humidity_on_pressure.data = np.full_like(
+        relative_humidity_on_pressure.data, humidity
+    )
+
+    result = HailSize()(
+        ccl_temperature,
+        ccl_pressure,
+        temperature_on_pressure_levels,
+        relative_humidity_on_pressure,
+    )
+
+    np.testing.assert_array_equal(result.data, expected)
+    metadata_check(result)
+
+
+def test_temperature_too_high(
+    temperature_on_pressure_levels,
+    ccl_pressure,
+    ccl_temperature,
+    relative_humidity_on_pressure,
+):
+    """Tests for the case where there are grid squares where the temperature
+    doesn't drop below 268.15K at any pressure."""
```

Comment:
Can you specify what happens in this case, i.e. returns zero hail size (rather than say a masked array).

Reply:
updated